src/main/java/com/librato/disco/Level2CacheStrategy.java

```diff
@@ -0,0 +1,29 @@
+package com.librato.disco;
+
+import org.apache.curator.framework.recipes.cache.ChildData;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public interface Level2CacheStrategy {
+    /**
+     * Signals to the strategy that these items were successfully found
+     * from a delegate cache.
+     */
+    void accept(List<ChildData> items);
```

Comment:
I guess where i'm coming from is that the implementor could use the delta between cached nodes and current nodes as a decision point on whether to promote or not without having to track extra state.